index.bs

```diff
@@ -8991,41 +9149,85 @@ input with a given set of file paths.
 
-The [=remote end steps=] given |session|, and |command parameters| are:
+The [=remote end steps=] given |command parameters| are:
 
 1. Let |context id| be the value of the <code>context</code> field of
    |command parameters|.
 
-1. Let |context| be the result of [=trying=] to [=get a browsing context=]
-   with |context id|.
+1. Let |context| be the result of [=trying=] to [=get a browsing context=] with
+   |context id|.
 
-1. Let |document| be |context|'s [=active document=].
+1. If |context| is not a [=top-level browsing context=], return [=error=] with
+   [=error code=] [=invalid argument=].
 
-1. Let |environment settings| be the [=environment settings object=] whose
-   [=relevant global object=]'s <a>associated <code>Document</code></a> is
-   |document|.
+1. Let |element| be the [=current file dialog=] for |context|.
 
-1. Let |realm| be |environment settings|' [=realm execution context=]'s
-   Realm component.
+1. If |element| is null, return [=error=] with [=error code=] [=no such file
+   dialog=]
 
-1. Let |reference| be the value of the <code>element</code> field of
-   |command parameters|.
+1. Let |files| be the value of the <code>files</code> field of |command parameters|.
 
-1. Let |element| be the result of [=trying=] to [=deserialize remote reference=]
-   with |reference|, |realm|, and |session|.
+1. Perform implementation-specific steps to [=update the file selection=] for
+   |element| with |files| as if a user selected |files| through a [=show the
+   picker|file picker=] prompt.
 
-1. If |element| doesn't implement {{HTMLInputElement}} return [=error=] with [=error code=]
-   [=unable to set file input=].
+   1. If the implementation is unable to [=update the file selection=], return
+      [=error=] with [=error code=] [=unable to set file input=].
 
-1. If the [=input type=] of |element| doesn't match <code>"file"</code> return [=error=] with [=error code=]
-   [=unable to set file input=].
+1. Return [=success=] with data null.
 
-1. Let |files| be the value of the <code>files</code> field of |command parameters|.
+</div>
 
-1. [=Update the file selection=] for |element| with |files| as if a user
-   selected |files| through the [=show the picker|file picker=] prompt.
+### Events ### {#module-input-events}
 
-1. Return [=success=] with data null.
+#### The input.fileDialogIntercepted Event #### {#event-input-fileDialogIntercepted}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl remote-cddl">
+         input.FileDialogIntercepted = (
+            method: "input.fileDialogIntercepted",
+            params: input.FileDialogInfo
+         )
+
+         input.FileDialogInfo = {
+            context: browsingContext.BrowsingContext
+         }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm>
+
+The [=remote end event trigger=] is the <dfn export>intercept WebDriver BiDi
+file dialog</dfn> steps given |context| and |element|.
+
+1. Assert: |context| is a [=top-level browsing context=].
```

Comment:
File dialogs can be not only from the top-level contexts, right?

Reply:
Indeed. I'll resolve this once I've made the change.